### PR TITLE
Added metrics cardinality handling for Google Cloud Spanner receiver - part 3.

### PR DESCRIPTION
**Pull request by ydrozhdzhal**
Final part to bind all together: algorithm, filter resolver metrics builder, receiver and config.

### CODE REVIEW

**Comment by ydrozhdzhal** [comment]
@MovieStoreGuy , @jpkrohling this is the final, 3d part.

File: receiver/googlecloudspannerreceiver/config_test.go
Comment (MovieStoreGuy) [inline receiver/googlecloudspannerreceiver/config_test.go:31]:
```suggestion
cardinalityLimit = 200000
```

keeping the naming simple :)

**Review by MovieStoreGuy — COMMENTED** [review]
So far it looks fine, be happy to approve once testify mock has been adopted, just simplifies the defined types.

File: receiver/googlecloudspannerreceiver/config_test.go
Comment (ydrozhdzhal) [reply receiver/googlecloudspannerreceiver/config_test.go]: Done.

**Comment by ydrozhdzhal** [comment]
@MovieStoreGuy , looks like that in this particular case using of mocks makes tests more complicated and it is hard to adopt.

**Review by asukhyy — APPROVED** [review]

**Review by kostia-hospodarysko — APPROVED** [review]

**Review by MovieStoreGuy — APPROVED** [review]
This looks solid.

**Comment by ydrozhdzhal** [comment]
@jpkrohling , @bogdandrutu , all reviews received, except maintainer's one. Can you look at this and approve and merge if everything is OK?

File: receiver/googlecloudspannerreceiver/README.md
Comment (jpkrohling) [inline receiver/googlecloudspannerreceiver/README.md:56]:
```suggestion
- **collection_interval** - this receiver runs periodically. Each time it runs, it queries Google Cloud Spanner, creates metrics, and sends them to the next consumer (default: 1 minute). **It is not recommended to change the default value of collection interval, since new values for metrics in the Spanner database appear only once a minute.**
```

File: receiver/googlecloudspannerreceiver/README.md
Comment (jpkrohling) [inline receiver/googlecloudspannerreceiver/README.md:59]:
```suggestion
- **cardinality_total_limit** - limit of active series per 24 hours period. If specified, turns on cardinality filtering and handling. If zero or not specified, cardinality is not handled. You can read [this document](cardinality.md) for more information about cardinality handling and filtering.
```

File: receiver/googlecloudspannerreceiver/config.go
Comment (jpkrohling) [inline receiver/googlecloudspannerreceiver/config.go:60]: What's this `s` at the end of the format? Also, try to use double quotes instead of backticks here.

File: receiver/googlecloudspannerreceiver/config_test.go
Comment (jpkrohling) [inline receiver/googlecloudspannerreceiver/config_test.go:31]: Tip: you can use underscores to make the number more readable.

```suggestion
	cardinalityLimit = 200_000
```

**Review by jpkrohling — COMMENTED** [review]
Looks good, just a couple of small changes to the docs.

File: receiver/googlecloudspannerreceiver/README.md
Comment (ydrozhdzhal) [reply receiver/googlecloudspannerreceiver/README.md]: Done.

File: receiver/googlecloudspannerreceiver/README.md
Comment (ydrozhdzhal) [reply receiver/googlecloudspannerreceiver/README.md]: Done.

File: receiver/googlecloudspannerreceiver/config_test.go
Comment (ydrozhdzhal) [reply receiver/googlecloudspannerreceiver/config_test.go]: Done.

File: receiver/googlecloudspannerreceiver/config.go
Comment (ydrozhdzhal) [reply receiver/googlecloudspannerreceiver/config.go]: Done. `s` - it was my mistake ((

**Comment by ydrozhdzhal** [comment]
@jpkrohling , addressed all comments, re-requested review. In case of merging this I will ask you the following: when squashing commits - can you leave only the main commit message and do not include messages from CR related commits? Other option is to allow me squash before merging.

**Comment by ydrozhdzhal** [comment]
Re-based by myself, to re-trigger flaky test jobs.

**Comment by jpkrohling** [comment]
Please, do not squash the commits between reviews, as it makes it harder to spot the differences between a code that has been reviewed already and the newest changes.

File: receiver/googlecloudspannerreceiver/config.go
Comment (jpkrohling) [inline receiver/googlecloudspannerreceiver/config.go:46]:
```suggestion
		return fmt.Errorf("%v %q must be not lower than 60s, current value is %v", config.ID(),
```

Comment (ydrozhdzhal) [reply receiver/googlecloudspannerreceiver/config.go:46]: In this particular case s means seconds

Comment (jpkrohling) [reply receiver/googlecloudspannerreceiver/config.go:46]: Got it. Thanks for the confirmation.

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**
